polaris-react/src/components/UnstyledButton/UnstyledButton.tsx

```diff
@@ -89,6 +90,7 @@ export function UnstyledButton({
         onKeyUp={onKeyUp}
         onKeyPress={onKeyPress}
         {...rest}
+        {...(disabled ? preventedInteraction : null)}
```

Comment:
Similar to the Button component, we should be doing the early return within a wrapping function here, something like:

```tsx
function handleClick(event: React.MouseEvent<HTMLButtonElement>) {
  if (disabled) {
    event.preventDefault();
    return;
  }
  onClick && onClick(event)
}

function handleKeyPress(event: React.KeyboardEvent<HTMLButtonElement>) {
  if (disabled) {
    event.preventDefault();
    return;
  }
  onKeyPress && onKeyPress(event)
}
```

and then update the methods passed to the <button> element to be `onKeyPress={handleKeyPress} onClick={handleClick}`. you'll need to explicitly destructure the `onClick` variable from the `props` object, too.